jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/AdministrationController.java

```diff
@@ -298,4 +307,28 @@ private void checkForAdminPermissions() {
         long forumId = componentService.getComponentOfForum().getId();
         componentService.checkPermissionsForComponent(forumId);
     }
+
+    private JsonResponse saveOrUpdateGroup(@Valid @RequestBody GroupAdministrationDto groupDto, BindingResult result, Locale locale) throws org.jtalks.common.service.exceptions.NotFoundException {
+        checkForAdminPermissions();
+        if (result.hasFieldErrors() || result.hasGlobalErrors()) {
+            return new JsonResponse(JsonResponseStatus.FAIL, result.getAllErrors());
+        }
+        try {
+            groupService.saveGroup(groupDto);
```

Comment:
To me DTO is the presentation layer (controllers) and services may not know how the results are transformed. But your arguments are compelling too. I'm okay to leave it as is, but let's discuss this again during the next meeting.